packages/core/src/internal/controllers/aria-popup-trigger.controller.spec.ts

```diff
@@ -33,16 +31,16 @@ describe('aria-popup-trigger.controller', () => {
 
   it('should initialize aria attributes for popup type triggers', async () => {
     await componentIsStable(component);
-    expect(component.getAttribute('aria-controls')).toBe('popup-el');
-    expect(component.getAttribute('aria-haspopup')).toBe('true');
-    expect(component.getAttribute('aria-expanded')).toBe('false');
+    expect(component.ariaControls).toBe('popup-el');
+    expect(component.ariaHasPopup).toBe('true');
+    expect(component.ariaExpanded).toBe('false');
   });
 
   it('should NOT initialize aria attributes for triggers without a popup id', async () => {
     await componentIsStable(noponent);
     expect(noponent.hasAttribute('popup')).toBe(false);
-    expect(noponent.hasAttribute('aria-controls')).toBe(false);
-    expect(noponent.hasAttribute('aria-haspopup')).toBe(false);
-    expect(noponent.hasAttribute('aria-expanded')).toBe(false);
+    expect(noponent.ariaControls).toBe(undefined);
+    expect(noponent.ariaHasPopup).toBe(null);
+    expect(noponent.ariaExpanded).toBe(null);
```

Comment:
We can now use aria properties directly.  In some cases like datagrid/treeview this seems to help performance as it enables the property to be set in AOM and let the browser manage updating the DOM